### PR TITLE
Replaced boost proeprty_map with rapidxml

### PR DESCRIPTION
**Pull request by UebelAndre**
First step to removing boost as a dependency. RapidXML is the what boost uses in it's `property_tree` so the performance should be pretty comparable. With boost removed as a dependency, compiling for all platforms becomes way simpler.

### CODE REVIEW

File: build/fetch_dependencies.py [hunk 1]
```diff
@@ -75,7 +75,8 @@
     ghRoot+"common-lib-ext-OpenGL.git"          : ["Common/Lib/Ext/OpenGL",     "master"],
     ghRoot+"common-lib-ext-tinyxml-2.6.2.git"   : ["Common/Lib/Ext/tinyxml",    "master"],
     ghRoot+"common-lib-ext-zlib-1.2.8.git"      : ["Common/Lib/Ext/zlib",       "master"],
-    "https://github.com/g-truc/glm.git"         : ["Common/Lib/Ext/glm",        "master"]
+    "https://github.com/g-truc/glm.git"         : ["Common/Lib/Ext/glm",        "master"],
+    "https://github.com/discord/rapidxml.git"   : ["External/rapidxml/rapidxml-submodule", "master"]
```
Comment (NPCompress) [inline build/fetch_dependencies.py:79]: Please use Common/Lib/Ext/rapidxml  path

File: Compressonator_Root.props [hunk 2]
```diff
@@ -16,6 +16,7 @@
     <Compressonator_OPENEXR>$(SolutionDir)..\..\..\..\Common\Lib\Ext\OpenEXR\openexr-2.2.0\</Compressonator_OPENEXR>
     <Compressonator_ILMBASE>$(SolutionDir)..\..\..\..\Common\Lib\Ext\OpenEXR\ilmbase-2.2.0\</Compressonator_ILMBASE>
     <Compressonator_OPENGL>$(SolutionDir)..\..\..\..\Common\Lib\Ext\OpenGL\</Compressonator_OPENGL>
+    <Compressonator_RAPIDXML>$(SolutionDir)..\..\..\External\rapidxml\</Compressonator_RAPIDXML>
```
Comment (NPCompress) [inline Compressonator_Root.props:19]: Please use ..\..\..\..\Common\Lib\Ext\rapidxml 

This source uses Boost and MIT licences, which is ok if used as user downloaded code in common.  MIT licenses is highly preferred.

Comment (UebelAndre) [reply Compressonator_Root.props:19]: I don't have the bandwidth to work on this project anymore. After the refactor to VS2017 I no longer know how to build and test changes. I had been working for a few months on proper CMake integration which added better cross platform support. In my opinion, if this project is intended to be a cross platform product then it needs to be using CMake and any Visual Studio artifacts should be deleted. I think you have the power to make these changes so if you're interested, I'm happy for you to do so, if not, you can decline this pull request.

My [feature/cpp17](https://github.com/UebelAndre/Compressonator/tree/feature/cpp17) branch has all the CMake migration work in it as well as MacOS support for the GUI as well. I'll leave it up for some time but will likely delete it in the future as it becomes more and more detached form HEAD of master. I think this project has a lot of potential but there's a lot of cleanup that needs to be done to make it maintainable.

**Comment by NPCompress** [comment]
@UebelAndre  This pull request will be merged and changes will be added as required. Your work is greatly appreciated. Removal of VS and use of only CMake is been fully adopted for the next release and your work has helped move this forward, The migration and cleanup work you did in feature/cpp17 branch will be reference and updated to accommodate the current directory structures for the upcoming release.

**Comment by UebelAndre** [comment]
@NPCompress thanks! That’s really nice to hear.

If you merge these changes I can open a PR for the larger set of changes and add comments to give a full context dump

**Pull request merged**